File: src/domain/types/advisor.ts
/*
 * @Description: 投顾推荐类型定义（金额字段存「分」）
 *
 * 与 Rust models/advisor.rs 对齐，字段名用 camelCase（serde 在 Rust 端做转换）。
 */

/** 推荐方向 */
export type AdvisorDirection = 'buy' | 'sell'

/** 投顾推荐信号 */
export type AdvisorSignal = {
  id: number
  advisor: string
  assetId: number
  direction: AdvisorDirection
  signalAt: string
  /** 推荐参考价（分） */
  refPrice: number
  /** 目标价（分，可空） */
  targetPrice: number | null
  /** 止损位（分，可空） */
  stopLoss: number | null
  /** 假设仓位（股数） */
  hypotheticalQty: number
  note: string | null
  createdAt: string
  updatedAt: string
  // JOIN 字段（来自 assets 表）
  assetCode?: string | null
  assetName?: string | null
  assetMarket?: string | null
}

export type AdvisorSignalCreatePayload = Omit<
  AdvisorSignal,
  'id' | 'createdAt' | 'updatedAt' | 'assetCode' | 'assetName' | 'assetMarket'
>

export type AdvisorSignalUpdatePayload = Omit<
  AdvisorSignal,
  'createdAt' | 'updatedAt' | 'assetCode' | 'assetName' | 'assetMarket'
>

/** 跟随 + 复盘记录 */
export type FollowUp = {
  id: number
  signalId: number
  followed: boolean
  actualPrice: number | null
  actualQty: number | null
  actualAt: string | null
  linkedTradeId: number | null
  reason: string | null
  rangeHigh: number | null
  rangeLow: number | null
  rangeEndClose: number | null
  reviewedAt: string | null
  createdAt: string
  updatedAt: string
}

export type FollowUpUpsertPayload = Omit<FollowUp, 'id' | 'createdAt' | 'updatedAt'>

/** 推荐方向标签 */
export const ADVISOR_DIRECTION_LABELS: Record<AdvisorDirection, string> = {
  buy: '买入推荐',
  sell: '卖出推荐',
}

File: src-tauri/src/models/advisor.rs
/*
 * @Description: 投顾推荐与复盘数据模型
 *
 * 金额字段（ref_price/target_price/stop_loss/actual_price/range_*）一律存「分」(i64)。
 * 数量字段（hypothetical_qty/actual_qty）存原始股数。
 */

use serde::{Deserialize, Serialize};

/// 投顾推荐信号（金额字段存「分」）
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AdvisorSignal {
    pub id: i64,
    pub advisor: String,
    pub asset_id: i64,
    pub direction: String,
    pub signal_at: String,
    pub ref_price: i64,
    pub target_price: Option<i64>,
    pub stop_loss: Option<i64>,
    pub hypothetical_qty: i64,
    pub note: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    // JOIN 出来的标的信息，便于前端展示
    pub asset_code: Option<String>,
    pub asset_name: Option<String>,
    pub asset_market: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateAdvisorSignalPayload {
    pub advisor: String,
    pub asset_id: i64,
    pub direction: String,
    pub signal_at: String,
    pub ref_price: i64,
    pub target_price: Option<i64>,
    pub stop_loss: Option<i64>,
    pub hypothetical_qty: i64,
    pub note: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateAdvisorSignalPayload {
    pub id: i64,
    pub advisor: String,
    pub asset_id: i64,
    pub direction: String,
    pub signal_at: String,
    pub ref_price: i64,
    pub target_price: Option<i64>,
    pub stop_loss: Option<i64>,
    pub hypothetical_qty: i64,
    pub note: Option<String>,
}

/// 跟随 + 复盘记录
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FollowUp {
    pub id: i64,
    pub signal_id: i64,
    pub followed: bool,
    pub actual_price: Option<i64>,
    pub actual_qty: Option<i64>,
    pub actual_at: Option<String>,
    pub linked_trade_id: Option<i64>,
    pub reason: Option<String>,
    pub range_high: Option<i64>,
    pub range_low: Option<i64>,
    pub range_end_close: Option<i64>,
    pub reviewed_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpsertFollowUpPayload {
    pub signal_id: i64,
    pub followed: bool,
    pub actual_price: Option<i64>,
    pub actual_qty: Option<i64>,
    pub actual_at: Option<String>,
    pub linked_trade_id: Option<i64>,
    pub reason: Option<String>,
    pub range_high: Option<i64>,
    pub range_low: Option<i64>,
    pub range_end_close: Option<i64>,
    pub reviewed_at: Option<String>,
}
